sunpy/coordinates/frames.py

```diff
@@ -0,0 +1,374 @@
+"""
+Common solar physics coordinate systems.
+
+This submodule implements various solar physics coordinate frames for use with
+the `astropy.coordinates` module.
+"""
+
+# NumPy import
+import numpy as np
+
+# Astropy imports
+from astropy import units as u
+from astropy.coordinates.representation import (CartesianRepresentation,
+                                                UnitSphericalRepresentation,
+                                                SphericalRepresentation)
+from astropy.coordinates.baseframe import (BaseCoordinateFrame,
+                                           RepresentationMapping)
+from astropy.coordinates import FrameAttribute
+
+# SunPy imports
+from sunpy import sun  # For Carrington rotation number
+from representation import (SphericalWrap180Representation,
+                            UnitSphericalWrap180Representation)
+
+from .frameattributes import TimeFrameAttributeSunPy
+
+RSUN_METERS = sun.constants.constant('radius').si.to(u.m)
+DSUN_METERS = sun.constants.constant('mean distance').si.to(u.m)
+
+__all__ = ['HeliographicStonyhurst', 'HeliographicCarrington',
+           'Heliocentric', 'Helioprojective', 'HelioprojectiveRadial']
+
+
+class HeliographicStonyhurst(BaseCoordinateFrame):
+    """
+    A coordinate or frame in the Stonyhurst Heliographic
+    system.
+
+    This frame has its origin at the solar centre and the north pole above the
+    solar north pole, and the zero line on longitude pointing towards the
+    Earth.
+
+    Parameters
+    ----------
+    representation: `~astropy.coordinates.BaseRepresentation` or None
```

Comment:
```
`None`
```

(basically None, True, and False can be included in single back-ticks as their references should resolve)